lab: Fix inconsistencies in chapter introductions

teodorjuravlea opened the pull request:
This is the fix I've suggested for #107.
I haven't been able to setup the Docker container to test the changes locally, so please check if I've changed everything correctly.
The 'Compute' lab chapter could use an introduction paragraph.

Alex-deVis commented on content/lab-setup/lab-setup.md:
```diff
@@ -0,0 +1,24 @@
+# Setting up the Lab environment
+
+If you have already cloned the repository, make sure it is updated.
+Run this command inside the repository clone directory:
```
```suggestion
```

Alex-deVis commented on content/lab-setup/lab-setup.md:
````diff
@@ -0,0 +1,24 @@
+Run this command inside the repository clone directory:
+
+```console
+git pull --rebase
````
```suggestion
cd operating-systems
git pull --rebase
```

Alex-deVis reviewed (changes requested):
I left a few nitpicks. Besides those, the path to `lab-setup.md` is wrong.

I think its overkill to name the section `lab-setup` and add a file with the same name. Let us call the directory `misc/`. @teodutu, please provide your pedantic view on this.

teodorjuravlea commented:
Thank you. I've added your improvements, and I've also renamed the directory to `misc/`, since I assume that has to change regardless.
Are you sure the path to `lab-setup` in `config.yaml` is incorrect? It seems correct to me, based on how the other sections are set.

teodutu commented on config.yaml:
```diff
@@ -55,14 +55,14 @@ docusaurus:
-          - Introduction:
-              location: content/chapters/software-stack/lab
-              name: content/introduction
+        - Setting up the Lab environment:
```
```suggestion
        - Setting up the Lab Environment:
```

teodutu commented on content/misc/lab-setup.md:
````diff
@@ -0,0 +1,24 @@
+```
+
+The command may fail if you have uncommitted changes.
+If that is the case, commit changes and retry.
````
Commit or stash and pop after `git pull`? I generally support commits, but last semester's experience showed me that maybe the simpler option is the best / fastest. @Alex-deVis, what do you say?

Alex-deVis replied:
Right, but only say `git stash` -> `git pull` -> `git stash pop`. Give one comprehensive option, not a bunch.

teodutu replied:
This will confuse people. I wouldn't give options. One way to rule them all.

Alex-deVis replied:
To sort this out add the inline suggestion and mark this as resolved.

teodutu commented on content/misc/lab-setup.md:
```diff
@@ -0,0 +1,24 @@
+The command may fail if you have uncommitted changes.
+If that is the case, commit changes and retry.
+
+If you haven't already cloned the repository, do it and enter the repository:
```
```suggestion
If you haven't already cloned the repository, do so and then enter the repository:
```

teodutu commented on content/misc/lab-setup.md:
````diff
@@ -0,0 +1,24 @@
+cd operating-systems
+```
+
+Navigate to the chapter lab directory:
+
+```console
+cd content/chapters/software-stack/lab/
+```
````
The Docker container defines some aliases to navigate more quickly [1]. I still think they're pointless, but maybe we should mention them here. @Alex-deVis would love this.

[1] https://github.com/open-education-hub/operating-systems/blob/b6a0cbdff828bdabfb7403a597e325c747e6bfbe/util/os-runner/config_terminal.sh#L21-L25

teodutu reviewed (changes requested):
Great changes! The path to the setup guide in `config.yaml` looks right. Address my comments and then we can merge this.

Alex-deVis commented on config.yaml:
```diff
@@ -55,14 +55,14 @@ docusaurus:
-              location: content/chapters/software-stack/lab
-              name: content/introduction
+        - Setting up the Lab environment:
+            location: content/misc
+            name: lab-setup
```
> Are you sure the path to lab-setup in config.yaml is incorrect? It seems correct to me, based on how the other sections are set.

`location` + `name` determine the path to the file. I think one example would solve this:

Software Stack: `locaction = content/chapters/software-stack/lab` and `name = content/overview`. If you look for the overview file's path in Software Stack is `content/chapters/software-stack/lab/content/overview.md`.

Using the same logic for lab-env: `location = content/misc` and `name = lab-setup` leads to the path `content/misc/lab-setup.md`, which is our file.

teodorjuravlea commented:
I've added the changes, mentioned all possible chapters to navigate to, and I've added the mention about aliases in a new commit. I've copied the alias descriptions directly from `config_terminal.sh`, so tell me if I should change its structure.
I'll add the info about uncommited changes as soon as you decide on the final version.

teodutu commented on content/misc/lab-setup.md:
````diff
@@ -0,0 +1,34 @@
+```
+
+The command may fail if you have uncommitted changes.
+If that is the case, commit changes and retry.
````
```suggestion
If that is the case, stash your changes, retry, and pop the stash:

```console
student@os:~/operating-systems$ git stash

student@os:~/operating-systems$ git pull --rebase

student@os:~/operating-systems$ git stash pop
```
```

The suggestion is a bit malformed due to the backticks interfering with each other.

teodutu commented on content/misc/lab-setup.md:
````diff
@@ -0,0 +1,34 @@
+If you haven't already cloned the repository, do so and then enter the repository:
+
+```console
+git clone https://github.com/open-education-hub/operating-systems
+cd operating-systems
````
```suggestion
student@os:~$ git clone https://github.com/open-education-hub/operating-systems

student@os:~$ cd operating-systems
```

teodutu commented on content/misc/lab-setup.md:
````diff
@@ -0,0 +1,34 @@
+Navigate to a chapter's lab directory:
+
+```console
+cd content/chapters/<chapter-name>/lab/
````
```suggestion
student@os:~/operating-systems$ cd content/chapters/<chapter-name>/lab/
```

teodutu commented on content/misc/lab-setup.md:
````diff
@@ -0,0 +1,34 @@
+If you have already cloned the repository, make sure it is updated:
+
+```console
+cd operating-systems
+git pull --rebase
````
```suggestion
student@os:~$ cd operating-systems

student@os:~/operating-systems$ git pull --rebase
```
It's better to leave an empty line between subsequent commands to make them more clearly distinguishable.

teodutu reviewed (changes requested):
I made suggestions about stashing changes and added a prompt to console snippets.

teodutu reviewed (approved):
:+1:
